Upgrade borsh unstable dependency to v1.0 and make it stable

bgeron opened the pull request:
I checked manually that even with borsh =v1.0.0 the tests pass.

KodrAus commented on src/lib.rs:
```diff
@@ -441,7 +441,7 @@ pub enum Variant {
 )]
 #[cfg_attr(
     all(uuid_unstable, feature = "borsh"),
```
```suggestion
    feature = "borsh",
```

The `uuid_unstable` gate should no longer be necessary now that `borsh` is stable.

KodrAus reviewed (commented):
Thanks @bgeron! It looks like `borsh` has been stable for a little while now, but still a huge props to the authors for pulling it all together 👏

bgeron commented:
> Thanks @bgeron! It looks like `borsh` has been stable for a little while now, but still a huge props to the authors for pulling it all together 👏

Very much agree! :clap: I'm not involved in borsh but I'm thinking to start using it.

- I changed more to make feature `borsh` stable, per your suggestion.

- I noticed that it wasn't possible to compile `uuid` with only the `borsh` feature because it didn't yet imply `dep:borsh-derive`, so I added that implication. Now a dummy program that depends on both uuid and borsh works. I'm not so experienced with Cargo feature implication but it seems to work.

KodrAus commented on Cargo.toml:
```diff
@@ -71,6 +71,8 @@ fast-rng = ["rng", "rand"]
 sha1 = ["sha1_smol"]
 md5 = ["md-5"]
 
+borsh = ["dep:borsh", "dep:borsh-derive"]
```
The fact that this works on our MSRV now is great :+1: There's so much I can clean up elsewhere with our dependencies.

KodrAus reviewed (approved):
Thanks @bgeron! This looks good to me.
